### PR TITLE
Raae 1200: Add index level stopwords support in configuration in index schema

### PR DESCRIPTION
**Pull request by nkanu17**
# Add Index-Level STOPWORDS Configuration Support

## Summary
This PR adds support for configuring stopwords at index creation time through the `IndexInfo.stopwords` field, enabling users to control which words are filtered during indexing rather than only at query time.
Related to issue: #432 
## Motivation
Previously, RedisVL had no way to configure index-level `STOPWORDS` for FT.CREATE, even though redis-py's `create_index()` API supports it. Users could only control stopwords at query time via `TextQuery.stopwords`, which is insufficient for use cases like:
- Searching for phrases containing common words (e.g., "Bank of Glasberliner")
- Custom stopwords lists for domain-specific applications
- Disabling stopwords entirely for precise text matching

### Core Implementation
- **`redisvl/schema/schema.py`**: Added `stopwords: Optional[List[str]]` field to `IndexInfo` class
  - `None` (default): Use Redis default stopwords (~300 words)
  - `[]` (empty list): Disable stopwords completely (STOPWORDS 0)
  - Custom list: Use specified stopwords

- **`redisvl/index/index.py`**: Updated `SearchIndex.create()` and `AsyncSearchIndex.create()` to extract and pass stopwords from schema to `create_index()`

- **`redisvl/redis/connection.py`**: Updated `convert_index_info_to_schema()` to parse stopwords from FT.INFO output, enabling `from_existing()` to preserve stopwords configuration

- **`redisvl/redis/utils.py`**: Updated `cluster_create_index()` and `async_cluster_create_index()` to accept stopwords parameter

### Warning System
Added validation in `BaseSearchIndex._validate_query()` to warn users when using query-time stopwords with index-level STOPWORDS 0, as this is counterproductive (indexes all words but filters them at query time).

### Tests
- **22 new tests** (8 unit tests for schema, 3 for FT.INFO parsing, 11 integration tests)
- All tests passing (981/981 unit tests, 1003/1003 integration tests)
- Comprehensive coverage of all three stopwords modes and edge cases

### Documentation
- **`docs/user_guide/11_advanced_queries.ipynb`**: Added 8 cells (22-29) with examples and explanations
- **`docs/stopwords_interaction_guide.md`**: Comprehensive guide explaining index-level vs query-time stopwords interaction
- Updated docstrings in `IndexInfo` class with YAML and dict examples

### Disable stopwords (STOPWORDS 0)
```python
schema = IndexSchema.from_dict({
    "index": {
        "name": "company-idx",
        "prefix": "company",
        "stopwords": []  # Disable stopwords
    },
    "fields": [{"name": "name", "type": "text"}]
})
index = SearchIndex(schema=schema)
index.create()

# Now "Bank of Glasberliner" searches work correctly
query = TextQuery(text="Bank of Glasberliner", text_field_name="name", stopwords=None)
results = index.query(query)

```

### CODE REVIEW

File: redisvl/index/index.py [hunk 1]
```diff
@@ -248,6 +249,30 @@ def _validate_query(self, query: BaseQuery) -> None:
                     "Vector field using 'flat' algorithm does not support EF_RUNTIME query parameter."
                 )
 
+        # Warn if using query-time stopwords with index-level STOPWORDS 0
+        if isinstance(query, TextQuery):
```
Comment (justin-cechmanek) [inline redisvl/index/index.py]: AggregateHybridQuery also has the stopwords param. Should be check for that query type also?

File: docs/api/query.rst [hunk 2]
```diff
@@ -61,6 +61,12 @@ TextQuery
    :show-inheritance:
    :exclude-members: add_filter,get_args,highlight,return_field,summarize
 
+.. note::
+   The ``stopwords`` parameter in :class:`TextQuery` controls query-time stopword filtering (client-side).
```
Comment (justin-cechmanek) [inline docs/api/query.rst:70]: TextQuery and also AggregateHybridQuery both have query-time stopwords

File: tests/integration/test_stopwords_integration.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        reconstructed_index = SearchIndex.from_existing(index.name, redis_client=client)
+
+        # Verify stopwords configuration was preserved
+        assert reconstructed_index.schema.index.stopwords == []
+
+    finally:
+        try:
+            index.delete(drop=True)
+        except Exception:
+            pass
```
Comment (justin-cechmanek) [inline tests/integration/test_stopwords_integration.py]: I don't understand why we need a try-finally block here

File: tests/integration/test_stopwords_integration.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+        reconstructed_index = SearchIndex.from_existing(index.name, redis_client=client)
+
+        # Verify stopwords configuration was preserved
+        assert set(reconstructed_index.schema.index.stopwords) == {"the", "a", "an"}
+
+    finally:
+        try:
+            index.delete(drop=True)
+        except Exception:
+            pass
```
Comment (justin-cechmanek) [inline tests/integration/test_stopwords_integration.py]: Same question; why do we need a try-finally block for these tests?

File: tests/unit/test_field_modifier_ordering.py [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+
+        # Verify INDEXMISSING comes before SORTABLE
+        if "INDEXMISSING" in remaining_args and "SORTABLE" in remaining_args:
+            assert remaining_args.index("INDEXMISSING") < remaining_args.index(
+                "SORTABLE"
+            )
+
+        # Verify SORTABLE comes before UNF
+        if "SORTABLE" in remaining_args and "UNF" in remaining_args:
+            assert remaining_args.index("SORTABLE") < remaining_args.index("UNF")
```
Comment (justin-cechmanek) [inline tests/unit/test_field_modifier_ordering.py]: What is mlp_commands?

File: tests/integration/test_field_modifier_ordering_integration.py [hunk 6]
```diff
@@ -0,0 +1,501 @@
...
+    ):
+        """Verify that NumericField does not have index_empty attribute.
+
+        INDEXEMPTY is only supported for TEXT and TAG fields according to
+        RediSearch documentation. NumericFieldAttributes should not have
+        an index_empty attribute.
+        """
+        import inspect
+
+        from redisvl.schema.fields import NumericFieldAttributes
```
Comment (justin-cechmanek) [inline tests/integration/test_field_modifier_ordering_integration.py:416]: We can probably move the imports out of the individual tests and have them at the test file level.

**Review by justin-cechmanek — COMMENTED** [review]
Looking really good. Just a few questions, and suggest we include AggregateHybridQuery to the stopwords modifications.

File: tests/integration/test_stopwords_integration.py (hunk 3 above)
Comment (nkanu17) [reply tests/integration/test_stopwords_integration.py]: Yea fixing this right now and using pytest fixtures.

File: tests/integration/test_stopwords_integration.py (hunk 4 above)
Comment (nkanu17) [reply tests/integration/test_stopwords_integration.py]: Yea fixing this right now and using pytest fixtures.

File: tests/unit/test_field_modifier_ordering.py (hunk 5 above)
Comment (nkanu17) [reply tests/unit/test_field_modifier_ordering.py]: Yea this is oversight, removing right now.

File: docs/api/query.rst (hunk 2 above)
Comment (nkanu17) [reply docs/api/query.rst:70]: Thank you, will make changes.

**Comment by nkanu17** [comment]
> Looking really good. Just a few questions, and suggest we include AggregateHybridQuery to the stopwords modifications.



> Bank of America
@rbs333 
Actually I removed references to it and should have been pushed but noted! Thanks!
